File: src/hasoftware-library/src/hasoftware/api/AnError.java
package hasoftware.api;

import hasoftware.cdef.CDEFMessage;

public class AnError {

    private final int _number;
    private final int _code;
    private final String _message;

    public AnError(int number, int code, String message) {
        _number = number;
        _code = code;
        _message = message;
    }

    public AnError(CDEFMessage cdefMessage) {
        _number = cdefMessage.getU8();
        _code = cdefMessage.getU32();
        _message = cdefMessage.getAsciiL();
    }

    public void encode(CDEFMessage cdefMessage) {
        cdefMessage.putU8(_number);
        cdefMessage.putU32(_code);
        cdefMessage.putAsciiL(_message);
    }

    public int getNumber() {
        return _number;
    }

    public int getCode() {
        return _code;
    }

    public String getMessage() {
        return _message;
    }
}

File: src.net/hasoftware-library/Api/AnError.cs
using hasoftware.Cdef;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace hasoftware.Api
{
    public class AnError
    {
        private readonly int _number;
        private readonly int _code;
        private readonly string _message;

        public AnError(int number, int code, string message)
        {
            _number = number;
            _code = code;
            _message = message;
        }

        public AnError(CdefMessage cdefMessage)
        {
            _number = cdefMessage.GetU8();
            _code = cdefMessage.GetU32();
            _message = cdefMessage.GetAsciiL();
        }

        public void Encode(CdefMessage cdefMessage)
        {
            cdefMessage.PutU8(_number);
            cdefMessage.PutU32(_code);
            cdefMessage.PutAsciiL(_message);
        }

        public int Nummber { get { return _number; } }
        public int Code { get { return _code; } }
        public string Message { get { return _message; } }
    }
}
